presto-main/src/main/java/com/facebook/presto/operator/aggregation/BindableAggregationFunction.java

```diff
@@ -116,15 +114,12 @@ public InternalAggregationFunction specialize(BoundVariables variables, int arit
         List<Type> inputTypes = boundSignature.getArgumentTypes().stream().map(x -> typeManager.getType(x)).collect(toImmutableList());
         Type outputType = typeManager.getType(boundSignature.getReturnType());
 
-        AggregationFunction aggregationAnnotation = definitionClass.getAnnotation(AggregationFunction.class);
-        requireNonNull(aggregationAnnotation, "aggregationAnnotation is null");
```

Comment:
This is legitimate sanity check. It shouldn't be deleted although the variable is not used below. However, the error message is not intuitive.

```
checkCondition(
    definitionClass.getAnnotation(AggregationFunction.class) != null,
    FUNCTION_IMPLEMENTATION_ERROR,
    format("... must be annotated with ...");
```